### PR TITLE
hack-fixed circular dependency

### PR DESCRIPTION
**Pull request by ArcanisCz**
### Brief description:
When using creep-tasks with Typescript (and webpack), coppied /src/creep-tasks to my project (as per instructions), but was unable to solve this particular circular-dependency. Project compiles just fine, but in-game it throws error.

Tried to modify creep-tasks with Overmind version, because Overmind works, but nothing successful.

When using your rollup config instead of webpack, it doesn't work ingame either. And it prints warning
`(!) Circular dependency: src\creep-tasks\utilities\initializer.ts -> src\creep-tasks\TaskInstances\task_attack.ts -> src\creep-tasks\Task.ts -> src\creep-tasks\utilities\initializer.ts` 
Basically, its only first circular dependency, theese cycles are present for each task type. And its breaking, when there is more than 1 task.
(for webpack, i used https://github.com/aackerman/circular-dependency-plugin which printed one hell of cycles)

### Added features:
Thic PR included somewhat hacky solution for the problem, as i was unable to solve it on the design level.

### Changes to existing features:

- None

### Bugfixes: 

- None

## Testing checklist:

- [ ] Codebase compiles with current `tsconfig` configuration
- [x] Deployed and tested changes

### CODE REVIEW

**Comment by ArcanisCz** [comment]
My config is there, if you want https://github.com/ArcanisCz/screeps

**Comment by bencbartlett** [comment]
Interesting. I've seen these types of circular import errors in Overmind before I switched to using rollup instead of webpack. I'd rather not clutter up the global namespace by moving the task instantiator there (although that was my previous solution in Overmind). I'm repackaging this as a typed npm module; hopefully that should fix the issue, and if not we can reconsider this.

**Comment by bencbartlett** [comment]
I've resolved these problems in the latest release of this module. It should also now work with your bot: I downloaded your `screeps` repository and removed the local `creep-tasks` directory, then ran `npm install creep-tasks`, modifying the imports to point to the npm module. It compiles and runs on the sim fine even without the CircularDependencyPlugin you added; however, you'll need to add `'.js'` to your resolve extensions in your webpack config to get it to work.

**Comment by ArcanisCz** [comment]
Great! But i am not sure about use-case with creep-tasks as a dependency. For it being dependency, its not extensible by custom tasks (they are hardcoded in initializeTask)

**Comment by bencbartlett** [comment]
Using the module with npm ensures that it correctly compiles and integrates into your code, but you are correct that it doesn't allow you to modify the module. If you'd like to extend or modify it, you should install the module from source like you did in your codebase previously. (Also, I've added almost all creep actions to this module as tasks; what functionality were you planning on implementing? Feel free to make a pull request if you'd like to get it integrated into this module!)

The circular dependency issues that you addressed are specific to the way your codebase (using webpack) bundles the compiled output. (For example, I use rollup and I don't experience this issue.) Given this, I don't think it's necessary to add a global fix for this issue to the module, but I will add a link to this issue in an FAQs page that I'll add to the wiki shortly.

**Comment by ArcanisCz** [comment]
Interesting fact, with your current tsconfig.json, it works on webpack without that hack! (coppied to my project with creep-tasks source)

I dont intend to add anything at the moment, since i am fresh new in screeps. But at some point, i will probably do. And some my own modifications and experimens, you know :D

By the way, many thanks for your articles, very much enjoyed the story. And for sharing creep-tasks. Copying whole AI is boring, but theese block are excellent, to use and to learn from them :)
